spec/models/location_spec.rb

```diff
@@ -0,0 +1,24 @@
+require "rails_helper"
+
+RSpec.describe "location" do
+  scenario "add a new location" do
+    back_bay = Location.new(body: 'Back Bay')
+    back_bay.save
+
+    expect(Location.all.length).to eq(1)
+  end
+  scenario "edits a location" do
+    back_bay = Location.new(body: 'Back Bay')
+    back_bay.save
+    back_bay.body = 'Not Back Bay'
+
+    expect(back_bay.body).to eq('Not Back Bay')
+  end
+  scenario "destroys a location" do
+    back_bay = Location.new(body: 'Back Bay')
+    back_bay.save
+    back_bay.destroy
+
+    expect(Location.all.length).to eq(0)
+  end
+end
```

Comment:
Should anyone be able to add a location, or just an authenticated user? Maybe an admin user?

Reply:
Just the admin.

Reply:
If that is the case, you'll want to create a User model that has the role of admin, or an Admin model for specific admin users. From there, you'll want to create a User in this tests and only have the new location be saved to the database if it is an admin who is creating it. This may require a join between User and Location

Reply:
They haven't pushed the Users migrations, models etc. So should I hold off on working on this and/or other things that are particular to admins, and users in general?

The users table is apparently generated by devise, but without admin stuff. Do they need to create a migration for adding that role?